include/seqan3/io/detail/misc_input.hpp

```diff
@@ -99,44 +99,57 @@ inline auto make_secondary_istream(std::basic_istream<char_t> & primary_stream,
     for (size_t i = 0 ; i < read_chars; ++i)
         primary_stream.unget();
 
-    std::string const extension = filename.extension().string();
+    std::string extension{};
+    if (filename.has_extension())
+        extension = filename.extension().string().substr(1);
 
     // set return value appropriately
     if (read_chars == magic_number.size() && contrib::_bgzfCheckHeader(magic_number.data())) // BGZF
     {
     #ifdef SEQAN3_HAS_ZLIB
-        if ((extension == ".gz") || (extension == ".bgzf"))
+        if ((std::ranges::find(gz_compression::file_extensions, extension) !=
+             std::ranges::end(gz_compression::file_extensions)) ||
+            (std::ranges::find(bgzf_compression::file_extensions, extension) !=
+             std::ranges::end(bgzf_compression::file_extensions)))
+        {
             filename.replace_extension();
+        }
 
         return {new contrib::basic_bgzf_istream<char_t>{primary_stream},
                 stream_deleter_default};
     #else
         throw file_open_error{"Trying to read from a bgzf file, but no ZLIB available."};
     #endif
     }
-    else if (starts_with(magic_number, magic_header<gz_compression>)) // GZIP
+    else if (starts_with(magic_number, gz_compression::magic_header)) // GZIP
     {
     #ifdef SEQAN3_HAS_ZLIB
-        if ((extension == ".gz") || (extension == ".bgzf"))
-            filename.replace_extension();
-
+        if ((std::ranges::find(gz_compression::file_extensions, extension) !=
+             std::ranges::end(gz_compression::file_extensions)) ||
+            (std::ranges::find(bgzf_compression::file_extensions, extension) !=
+             std::ranges::end(bgzf_compression::file_extensions)))
```

Comment:
Should we maybe add a `contains()` free function? :see_no_evil:

Reply:
not a free function but I'll dry it into a lambda